### PR TITLE
 Update release workflow instructions to add details about `conda-forge` releases

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
# Description

See #3442 for more details; this PR makes a small documentation update to ensure that the conda-forge release can be tracked and that no discrepancies occur between the PyPI and conda distributions of PyBaMM.

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: .github/release_workflow.md [hunk 1]
```diff
@@ -77,3 +77,5 @@ Some other essential things to check throughout the release process -
   git tag -f <tag_name> <commit_hash>
   git push -f <pybamm-team/PyBaMM_remote_name> <tag_name>  # can only be carried out by the maintainers
   ```
+- If changes are made to the API, console scripts, entry points, new optional dependencies are added, support for major Python versions is dropped or added, or core project information and metadata are modified at the time of the release, make sure to update the `meta.yaml` file in the `recipe/` folder of the [conda-forge/pybamm-feedstock](https://github.com/conda-forge/pybamm-feedstock) repository accordingly by following the instructions in the [conda-forge documentation](https://conda-forge.org/docs/maintainer/updating_pkgs.html#updating-the-feedstock-repository) and re-rendering the recipe
+- The conda-forge release workflow will automatically be triggered following a stable PyPI release, and the aforementioned updates can be carried out either in a personal fork of the feedstock repository, or directly in the main repository by pushing changes to the automated PR created by the conda-forge-bot.
```
Comment (Saransh-cpp) [inline .github/release_workflow.md:81]:
```suggestion
- The conda-forge release workflow will automatically be triggered following a stable PyPI release, and the aforementioned updates should be carried out directly in the main repository by pushing changes to the automated PR created by the conda-forge-bot. A manual PR can also be created if the updates are not included in the automated PR for some reason. This manual PR must bump the build number in `meta.yaml`.
```

**Review by Saransh-cpp — APPROVED** [review]
Looks good, thanks @agriyakhetarpal! :)

File: .github/release_workflow.md (hunk 1 above)
Comment (agriyakhetarpal) [reply .github/release_workflow.md:81]: This information is given in the link to the conda-forge documentation

Comment (agriyakhetarpal) [reply .github/release_workflow.md:81]: Adding it as per your suggestion because it would help as a failsafe

**Pull request merged**